hedera-node/src/main/java/com/hedera/services/ledger/interceptors/StakeAwareAccountsCommitsInterceptor.java

```diff
@@ -0,0 +1,319 @@
+			final int num,
+			final EntityChangeSet<AccountID, MerkleAccount, AccountProperty> pendingChanges,
+			final Set<Long> hasBeenRewarded,
+			final long latestEligibleStart) {
+		int changesSize = pendingChanges.size();
+		final var account = pendingChanges.entity(num);
+		final var changes = pendingChanges.changes(num);
+
+		final var curStakeeNum = (account != null) ? account.getStakedId() : 0L;
+		final var newStakeeNum = manager.getAccountStakeeNum(changes);
+
+		if (curStakeeNum > 0 && curStakeeNum != newStakeeNum) {
+			// Stakee has been replaced, withdraw initial balance from ex-stakee
+			final var exStakeeI = findOrAdd(curStakeeNum, pendingChanges);
+			manager.updateStakedToMe(exStakeeI, -account.getBalance(), pendingChanges);
+			if (exStakeeI == changesSize) {
+				changesSize++;
+			} else if (!hasBeenRewarded.contains(curStakeeNum)) {
+				payRewardIfRewardable(pendingChanges, exStakeeI, hasBeenRewarded, latestEligibleStart);
+				// do we need this ?? since stakedToMe is only for account stakes
+			}
+		}
+		if (newStakeeNum > 0) {
+			// Add pending balance to new stakee
+			final var newStakeeI = findOrAdd(newStakeeNum, pendingChanges);
+			manager.updateStakedToMe(newStakeeI, finalBalanceGiven(account, changes), pendingChanges);
+			if (newStakeeI == changesSize) {
+				changesSize++;
+			} else if (!hasBeenRewarded.contains(newStakeeNum)) {
+				payRewardIfRewardable(pendingChanges, newStakeeI, hasBeenRewarded, latestEligibleStart);
+				// do we need this ?? since stakedToMe is only for account stakes
+			}
+		}
+		return changesSize;
+	}
+
+	void payRewardIfRewardable(
+			final EntityChangeSet<AccountID, MerkleAccount, AccountProperty> pendingChanges,
+			final int stakeeI,
+			final Set<Long> hasBeenRewarded,
+			final long latestEligibleStart) {
+		final var account = pendingChanges.entity(stakeeI);
+		final var changes = pendingChanges.changes(stakeeI);
+		final var accountNum = pendingChanges.id(stakeeI).getAccountNum();
+		if (isRewardable(account, changes, latestEligibleStart)) {
+			payReward(account, changes, accountNum, hasBeenRewarded);
+		}
+	}
+
+	private void payReward(final MerkleAccount account,
+			final Map<AccountProperty, Object> changes,
+			final long accountNum,
+			final Set<Long> hasBeenRewarded) {
+		final var reward = rewardCalculator.updateRewardChanges(account, changes);
+		sideEffectsTracker.trackRewardPayment(accountNum, reward);
+		hasBeenRewarded.add(accountNum);
+	}
+
+	int findOrAdd(
+			final long accountNum,
+			final EntityChangeSet<AccountID, MerkleAccount, AccountProperty> pendingChanges
+	) {
+		final var n = pendingChanges.size();
+		for (int i = 0; i < n; i++) {
+			if (pendingChanges.id(i).getAccountNum() == accountNum) {
+				return i;
+			}
+		}
+		// This account wasn't in the current change set
+		pendingChanges.include(
+				STATIC_PROPERTIES.scopedAccountWith(accountNum),
+				accounts.get().get(EntityNum.fromLong(accountNum)),
+				Map.of());
```

Comment:
⚠️ `Map.of()` returns an immutable map; we need new `EnumMap<AccountProperty, Object>(AccountProperty.class)`.

Reply:
Ah yes ! 😞